### ISSUE TITLE
ExpansionPanelList merge the header semantics when it is not necessart 

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#33774]
## Steps to Reproduce

```
import 'package:flutter/material.dart';
import 'package:flutter/gestures.dart';
class ShoppingBasket extends StatefulWidget {
  @override
  ShoppingBasketState createState() => new ShoppingBasketState();
}

class MyItem {
  MyItem({ this.isExpanded: false, this.header, this.body });

  bool isExpanded;
  final String header;
  final String body;
}

class ShoppingBasketState extends State<ShoppingBasket> {

  bool expanded =false;
  @override
  Widget build(BuildContext context) {
    return new ListView(
      children: [
        new ExpansionPanelList(
          expansionCallback: (int index, bool isExpanded) {
            setState(() {
              expanded = !expanded;
            });
          },
          children: [
            new ExpansionPanel(
              canTapOnHeader: false,
              headerBuilder: (BuildContext context, bool isExpanded) {
                return new Text.rich(TextSpan(text:'head1' , recognizer: new TapGestureRecognizer()..onTap = () => print('Tap Here onTap')));
              },
              isExpanded: expanded,
              body: new Container(
                child: new Text('body1'),
              ),
            ),
          ],
        ),
      ],
    );
  }
}

void main() {
  runApp(new MaterialApp(
    home: new Scaffold(
      appBar: new AppBar(
        title: new Text('ExpansionPanel Example'),
      ),
      body: new ShoppingBasket(),
    ),
  ));
}
```

1. run the app with accessibility enabled

## Logs

```
flutter: ══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
flutter: The following assertion was thrown during a scheduler callback:
flutter: 'package:flutter/src/semantics/semantics.dart': Failed assertion: line 2499 pos 16: 'node.parent ==
flutter: null || !node.parent.isPartOfNodeMerging || node.isMergedIntoParent': is not true.
flutter:                                                                
flutter: Either the assertion indicates an error in the framework itself, or we should provide substantially
flutter: more information in this error message to help you determine and fix the underlying cause.
flutter: In either case, please report this assertion by filing a bug on GitHub:
flutter:   https://github.com/flutter/flutter/issues/new?template=BUG.md
flutter:                                                                
flutter: When the exception was thrown, this was the stack:             
flutter: #2      SemanticsOwner.sendSemanticsUpdate (package:flutter/src/semantics/semantics.dart:2499:16)
flutter: #3      PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:961:23)
flutter: #4      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:350:19)
flutter: #5      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:701:13)
flutter: #6      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:285:5)
flutter: #7      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1012:15)
flutter: #8      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:952:9)
flutter: #9      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:773:7)
flutter: #18     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:382:19)
flutter: #19     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:416:5)
flutter: #20     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:171:12)
flutter: (elided 10 frames from class _AssertionError, package dart:async, and package dart:async-patch)
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
flutter: Another exception was thrown: 'package:flutter/src/semantics/semantics.dart': Failed assertion: line 2499 pos 16: 'node.parent == null || !node.parent.isPartOfNodeMerging || node.isMergedIntoParent': is not true.
Syncing files to device iPhone X...                              1,349ms
```
If canTapOnHeader is false, there is no reason we force merge the header, user can still want to have their custom gesturerecognizer in the header.
